WebAPI/Controllers/CategoriesController.cs

```diff
@@ -22,4 +22,25 @@ public async Task<ActionResult<List<CategoryDetailDto>>> GetAll()
 
         return Ok(result);
     }
+
+    [HttpGet]
+    public async Task<ActionResult<string>> GetImage(Guid id)
+    {
+        var result = await _categoryService.GetByIdAsync(id);
+
+        return Ok(result.ImagePath);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<Task>> UpdateImage(CategoryDetailDto updatedCategory)
+    {
+        var result = _categoryService.UpdateAsync(updatedCategory);
```

Comment:
Where is `await`? Your original task context will not be able to take control after task will be executed asynchronous

Same case for `ProductsController`

Reply:
Because I want to check the result if it is successfull or not. But if I put await there, I can not assign to a variable (which is `result` here). Could you suggest me some way?

Reply:
Usually, in enterprise projects (I suggest we can use the same approach) if operation is successful or not is not checked directly in endpoints: if there was some issue on Business/Infrastructure layers that restricts future work, than you can easily throw an exception and specific middleware (we implemented it earlier) will take responsibility for providing error response. If we follow this approach, than you controller should return only one result - successful.

Reply:
Maybe I can implement after I completed the mentoring program?